### PR TITLE
MGMT-13471: SNO installation with OCP 4.13 lacks proper dnsmasq configuration

### PR DESCRIPTION
**Pull request by tsorya**
MGMT-13471: SNO installation with OCP 4.13 lacks proper dnsmasq configuration

## List all the issues related to this PR

- [ ] New Feature 
- [ ] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [x] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: internal/network/manifests_generator.go [hunk 1]
```diff
@@ -400,6 +402,12 @@ func createDnsmasqForSingleNode(log logrus.FieldLogger, cluster *common.Cluster)
 		"UNMANAGED_RESOLV_CONF": base64.StdEncoding.EncodeToString([]byte(unmanagedResolvConf)),
 	}
 
+	// Default is less then 4.13, no need to check error, if version not set no need to add additional params
```
Comment (osherdp) [inline internal/network/manifests_generator.go:405]:
```suggestion
	// Default is less than 4.13, no need to check error, if version not set no need to add additional params
```

File: internal/network/manifests_generator.go [hunk 2]
```diff
@@ -141,12 +141,14 @@ spec:
...
 
             [Service]
-            ExecStart=/usr/sbin/dnsmasq -k
+            ExecStart=/usr/sbin/dnsmasq -k {{.DNSMASQ_ADDITIONAL_PARAMS}}
 
             [Install]
             WantedBy=multi-user.target
 `
 
+const additionalDNSMasqParams = "-a 0.0.0.0 -a ::"
```
Comment (osherdp) [inline internal/network/manifests_generator.go:150]: my understanding is that ``::`` is just one address whereas ``::/0`` includes any address in the space of ipv6 addresses
any way, it's not critical. we'll just make sure to test the ipv6 case

Comment (tsorya) [reply internal/network/manifests_generator.go:150]: Yep, it is 0.0.0.0 in ipv6
Don't see why not to have same behavior as we have now

File: internal/network/manifests_generator.go [hunk 3]
```diff
@@ -400,6 +411,12 @@ func createDnsmasqForSingleNode(log logrus.FieldLogger, cluster *common.Cluster)
 		"UNMANAGED_RESOLV_CONF": base64.StdEncoding.EncodeToString([]byte(unmanagedResolvConf)),
 	}
 
+	// Default is less than 4.13, no need to check error, if version not set no need to add additional params
+	rhel9Version, _ := common.VersionGreaterOrEqual(cluster.OpenshiftVersion, "4.13.0-0.0")
```
Comment (osherdp) [inline internal/network/manifests_generator.go:415]: just wanted to make sure (you can also add unit-tests for those cases),
it works for both pre-release 4.13 versions (which I see you tested) and ``4.13.0``?

Comment (tsorya) [reply internal/network/manifests_generator.go:420]: you mean to test version greater? To say the truth i assumed it was tested already and not require retesting each time we use it. 
As far as i understand 0-0.0 means the minimal 4.13 and any version of 4.13 will hit it

**Review by osherdp — APPROVED** [review]

**Review by osherdp — APPROVED** [review]

**Pull request merged**